s2fft/utils/quadrature_torch.py

```diff
@@ -0,0 +1,256 @@
+import torch
+from s2fft.sampling import s2_samples as samples
+
+
+def quad_weights_transform(
+    L: int, sampling: str = "mwss", nside: int = 0
+) -> torch.tensor:
+    r"""Compute quadrature weights for :math:`\theta` and :math:`\phi`
+    integration *to use in transform* for various sampling schemes. Torch implementation of
+    :func:`~s2fft.quadrature.quad_weights_transform`.
+
+    Quadrature weights to use in transform for MWSS correspond to quadrature weights
+    are twice the base resolution, i.e. 2 * L.
+
+    Args:
+        L (int): Harmonic band-limit.
+
+        sampling (str, optional): Sampling scheme.  Supported sampling schemes include
+            {"mwss", "dh", "healpix}.  Defaults to "mwss".
+
+        nside (int, optional): HEALPix Nside resolution parameter.  Only required
+            if sampling="healpix".  Defaults to None.
+
+    Raises:
+        ValueError: Invalid sampling scheme.
+
+    Returns:
+        torch.tensor: Quadrature weights *to use in transform* for sampling scheme for
+        each :math:`\theta` (weights are identical as :math:`\phi` varies for given
+        :math:`\theta`).
+    """
+
+    if sampling.lower() == "mwss":
+        return quad_weights_mwss_theta_only(2 * L) * 2 * torch.pi / (2 * L)
```

Comment:
Can you add some extra tests catch these and following cases (see codecov comments).